### DISCUSSION TITLE
Horizontal mouse wheel scrolling?

### DISCUSSION THREAD

**Discussion by jesperlandberg** [idiotWu/smooth-scrollbar#355] [Q&A] [answered]
Not an issue, but just wanted to ask if this plugin supports horizontal scroll with mouse wheel? Or any examples on how to implement this.

Thanks for a great plugin.

**Comment by dolphin-wood** [comment]
Horizontal scrolling is supported in most environments. If you can't do so, it means the input of your mouse wheel is not standard thus we are not able to get `deltaX` value from it.

You can refer to [this module](https://github.com/idiotWu/smooth-scrollbar/blob/develop/src/utils/get-delta.js) to see how it handles wheel event.

**Comment by jesperlandberg** [comment]
@idiotWu Alright, but I am not able to do something like this http://paack.fr/ out of the box?

**Comment by dolphin-wood** [comment]
Ah, I took a look at the minified source code and found something interesting:

```js
void (void 0 === window.SCROLL_DIRECTION || "vertical" === window.SCROLL_DIRECTION ? t.__addMovement(a, l, !0) : t.__addMovement(l, a, !0)))
```

It seems that the author modified the wheel event handler in [lib/events/wheel.js](https://github.com/idiotWu/smooth-scrollbar/blob/develop/lib/events/wheel.js).

**Comment by sndrgb** [comment]
@jesperlandberg I have the same question, did you resolve with some tricks?
I don't think it's a good idea edit the source code directly.

**Comment by jesperlandberg** [comment]
Nope, By the same reason u didn't. Would be nice to see an option for this tho=)

**Comment by dolphin-wood** [comment]
@jesperlandberg @sndrgb well I think it's VERY weird to perform a horizontal scrolling while I scroll the wheel vertically, I'm afraid there's not going to be an option for that :(

I would suggest you make your own fork and use it.

**Comment by dolphin-wood** [comment]
Just thinking about adding a `reducer` concept which may solve this issue, check #84

**Comment by dolphin-wood** [comment]
Hey there! [8.0.0](https://github.com/idiotWu/smooth-scrollbar/releases/tag/v8.0.0) is out! Any feedback would be greatly appreciated 🙌 .

**Comment by oguilleux** [comment, marked as answer]
Just dropping the solution that helped me do horizontal scroll here for the future :

https://codepen.io/idiotWu/pen/KJNYye
